File: src/main/java/pt/dei/springmvcangularjs/controller/MidSController.java
package pt.dei.springmvcangularjs.controller;

import java.util.Map;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseBody;

import pt.dei.insulin_calculator_server.ws_manager.WebServicesManager;
import pt.dei.springmvcangularjs.models.MidsModel;
import pt.dei.springmvcangularjs.models.Response;

@Controller
@RequestMapping("/api/v1/mids")
public class MidSController {


    @Autowired
    private static WebServicesManager wsServicesManager;

    public MidSController() {
    	wsServicesManager = new WebServicesManager();
	}

    @RequestMapping("/")
    public @ResponseBody Response getMids(@RequestParam Map<String,String> params){
    	
    	MidsModel midsObject = new MidsModel(params.get("input_a"), params.get("input_b"), params.get("input_c"), params.get("input_d"), params.get("input_e"));

    	if (!midsObject.validate()){
    		return new Response();
    	}
    	
    	Response response = wsServicesManager.execute(midsObject);
    	//call cenas
    	return response;

    }

    @RequestMapping("/layout")
    public String getMidsPartialPage() {
        return "mids/layout";
    }




}

File: src/main/webapp/resources/js/controllers/MidsController.js

'use strict';


/**
 * CarController
 * @constructor
 */
var MidsController = function($scope, $http) {

    $scope.values = {
        input_a: '',
        input_b: 12,
        input_c: '',
        input_d: '',
        input_e: 50,
    };


    //get data from server using data sended
    $scope.getMids = function(data) {

        $http.get('api/v1/mids/', {params:$scope.values}).success(function(results){
            $scope.results = results;
            console.log(results);
        });


        $scope.values = {
                input_a: '',
                input_b: 12,
                input_c: '',
                input_d: '',
                input_e: 50,
        };


    };

    $scope.isThisDisabled=function(){

        for(var i in $scope.values)
        {
            if(!$scope.values[i])
                return true;
        }
       return false;
   }

};
